lib/contracts/contracts.go

```diff
@@ -0,0 +1,130 @@
+package contracts
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/omni-network/omni/halo/genutil/evm/predeploys"
+	"github.com/omni-network/omni/lib/netconf"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/params"
+
+	"cosmossdk.io/math"
+)
+
+// Contract defines a contract to monitor.
+type Contract struct {
+	Name               string
+	Address            common.Address
+	OnlyOmniEVM        bool
+	NotOmniEVM         bool
+	FundThresholds     *FundThresholds
+	WithdrawThresholds *WithdrawThresholds
+}
+
+// All returns all contracts for the given network relevant to the monitor.
+func All(ctx context.Context, network netconf.ID) ([]Contract, error) {
```

Comment:
I know I suggested the name `All`, but do you think we should rename to `ToMonitor()`? because this is really just a list of contracts to monitor. it does not include all contracts